AI revisions and blog content additions (#192)
fivetran/fivetran_connector_sdk: ai_and_connector_sdk/claude/pokeapi_tutorial/pokeapi_connector_tutorial.md

This tutorial is intended to be read in tandem with our [Building a Fivetran Connector in 1 Hour with Claude AI](https://www.fivetran.com/blog/building-a-fivetran-connector-in-1-hour-with-anthropics-claude-ai) blog post that was published in March 2024. You will build a **Fivetran Connector** that automatically syncs Pokemon data from the PokeAPI. You'll learn how we:

- Setup Claude to run in our terminal
- Cloned the Fivetran Connector SDK Examples to provide context
- Worked with and prompted Claude
- Itterated and tested the Connector code Claude generated
- Deployed the connector and saw it running int production

**Perfect for**: Data engineers, analysts, and developers who want to try out creating connectors with AI.

## Prerequisites

### Technical Requirements
- **Python 3.9 - 3.12+** installed on your machine
- **Basic Python knowledge** (functions, dictionaries, loops)
- **Fivetran Connector SDK** (we'll install this)

### API Access
- **PokeAPI** (completely free, no authentication required)
  - Public API endpoint: https://pokeapi.co/api/v2/pokemon/
  - No rate limits for basic usage
  - Comprehensive Pokemon data available

### Development Environment

> - **Access to Claude** (In terminal like the blog, in IDE chat or independant web access)

Reviewer: ```suggestion
- **Access to Claude** (in terminal like the blog, in IDE chat, or independent web access)
```

- **Terminal/Command line** access
- **Internet connection** for API calls

## Connector SDK Files

> In this example 2 or the 3 core files got generated to create this Connector SDK:

Reviewer: ```suggestion
In this example, two of the three core files are generated to create this Connector SDK:
```

### 1. `connector.py` - Main Connector Logic
```python
# This is your main connector file
# Contains all the Pokemon API connecion logic
# Handles data fetching, processing, and passing to Fivetran
```

### 2. `config.json` - Settings
```json
{
    "limit": "100",
    "request_delay": "1.0",
    "api_base_url": "https://pokeapi.co/api/v2"
}
```

### 3. `requirements.txt` - Dependencies

> no additional dependencies for this API are needed so this file was correctly skipped by the AI agent.

Reviewer: ```suggestion
No additional dependencies for this API are needed, so this file was correctly skipped by the AI agent.
```

## Context and The Prompt

> In this case we provided context by copying and pasting the [Fivetran Connector SDK technical reference](https://fivetran.com/docs/connector-sdk/technical-reference)

Reviewer: ```suggestion
In this case, we provided context by copying and pasting the [Fivetran Connector SDK technical reference](https://fivetran.com/docs/connector-sdk/technical-reference).
```

> we also generated this [Claude.md](https://github.com/fivetran/fivetran_connector_sdk/tree/main/ai_and_connector_sdk/claude/CLAUDE.md)

Reviewer: ```suggestion
We also generated this [Claude.md](https://github.com/fivetran/fivetran_connector_sdk/tree/main/ai_and_connector_sdk/claude/CLAUDE.md) file.
```

Here's the original prompt that created this connector:

> I’d like you to help me create and test a Fivetran connector SDK connector to gather lots of data from the PokéAPI and deliver that into my destination in Fivetran.

Then
> now I want to test the connector

> That was it for this case. We weren't particularly bothered by what data was sync'ed or it schema. In this exploration 4 tables were successfully created and populated.

Reviewer: ```suggestion
That was all in this case. We weren't particularly concerned about which data was synced or its schema. In this exploration, four tables were successfully created and populated.
```

> ## How to try out the claude code created during the blog

Reviewer: ```suggestion
## How to try out the Claude code created during the blog
```

1. **Create your project folder**:
   ```bash
   mkdir pokemon-connector
   cd pokemon-connector
   ```

> 2. **Copy the two files** above into your folder

Reviewer: ```suggestion
2. **Copy the two files** above into your folder: <br>
```

   You don't need the state.json and warehouse.db from the files/ directory they are autogenerated when you run `fivetran debug` for local testing and are included for completeness and independent inspection. 

3. **Test the connector**:
   ```bash
   fivetran debug --configuration config.json
   ```

6. **Check your data**:
   ```bash
   duckdb files/warehouse.db ".tables"
   duckdb files/warehouse.db "SELECT * FROM tester.pokemon"
   ```

## Next Steps

> Once you've got the connector claude generated for us working try using AI to enhance or change it

Reviewer: ```suggestion
Once you've got the connector Claude generated for us working, try using AI to enhance or change it:
```
